### ISSUE TITLE
[Workflow] Linkchecker improvements

### ISSUE THREAD

**Issue by gilzow** [platformsh/platformsh-docs#3240]
### Where on docs.platform.sh should be changed?

Scheduled link checker

### What exactly should be updated?

This is mainly for me to have a record for future updates to the schedule linkchecker.

The linkchecker application does [support adding your own Logger class](https://linkchecker.github.io/linkchecker/faq.html), so we could theoretically add a Logger class for markdown and do that instead of exporting to xml and then converting from xml --> markdown

The linkchecker application also [supports a configuration file](https://linkchecker.github.io/linkchecker/man/linkcheckerrc.html#) that exposes more options than the command line, one of which is the ability to ignore certain warnings. Instead of excluding github urls as we are doing now, we could instead ignore `http-rate-limited` which is what we're running into with GitHub. It does *not* have the ability to ignore 403 responses so we would still need to deal with that separately. However, we would need to figure out how to retrieve the configuration file since right now we're not checking out this repository when we run the workflow. A separate config file would also give us some flexibility in how we manage URLs we want to exclude.  

While linkchecker doesnt support ignoring specific server responses, it _does_ however support plugins, so it's possible we could write our own plugin giving us the ability to ignore certain server responses.

**Comment by gilzow** [comment]
Per damz we should go ahead and ignore https://support.blackfire.io/

**Comment by AnouckColson** [comment]
@gilzow Agreed, the Blackfire support link will keep raising false positives (cf. https://github.com/platformsh/platformsh-docs/issues/3249), so if there's a way to ignore it, that would be awesome.

**Comment by gilzow** [comment]
implemented linkchecker config file in #4399 . The xml --> markdown is working so as of now, the ROI on writing our own logger class is low. Closing as no longer applicable.
